src/main/java/net/getsentry/gocd/webhooknotifier/utils/Http.java

```diff
@@ -35,21 +51,79 @@ public static void PingWebhooks(PluginRequest pluginRequest, String type, Object
       return;
     }
 
-    String[] urls = ps.getTrimmedWebhookURLs();
-    for (int i = 0; i < urls.length; i++) {
-        try {
-          Post(urls[i], responseJsonStr);
-        } catch (Exception e) {
-          System.out.printf("    😺 failed to post request to %s: %s\n", urls[i], e.getMessage());
+    URI[] uris = ps.getWebhookURIs();
+    for (URI uri : uris) {
+      try {
+        List<Header> headers = new ArrayList<Header>();
+        String url = removeURIUserInfo(uri).toString();
+        String authToken = getAuthToken(uri, client);
+        if (authToken != null) {
+          headers.add(new BasicHeader(HttpHeaders.AUTHORIZATION, "Bearer " + authToken));
         }
```

Comment:
That check is redundant since the function short circuits and returns null if the userInfo is null. I went with this approach as I felt it better encapsulates the logic for obtaining the auth token, which includes the validation of preconditions like the presence of userInfo.